site/profile/manifests/device_manager.pp

```diff
@@ -0,0 +1,39 @@
+class profile::device_manager (
+  Hash $defaults = lookup('profile::device_manager::defaults'),
+  Hash $devices = lookup('profile::device_manager::devices'),
+) {
+
+  $devices.each |$device, $parameters| {
+    # Only manage devices that are configured for this node
+    if $parameters[proxy] == $trusted['certname'] {
+      # Look for node specific values, otherwise find default
+      $type = $parameters['type'] ? {
+        /(.*)/  => "${0}",
+        default => $defaults['type'],
+      }
+      $username = $parameters['username'] ? {
+        /(.*)/  => "${0}",
+        default => $defaults['username'],
+      }
```

Comment:
This is a strange looking construct. If the value is a String (any string, including an empty string) it will be the result otherwise `$defaults['username']`.
Is empty string matching intended?

Here is an idea to make it less clever:
```
$username = if $parameters['username'] =~ String { $parameters['username'] } else { $defaults['username'] }
```
Which can be turned into a function:
```
function mymodule::string_or_default($name, $parameters, $defaults) {
  if $parameters[$name] =~ String { $parameters[$name] } else { $defaults[$name] }
}
```
And used as:
```
$username        = mymodule::string_or_default('username', $parameters, $defaults)
$password        = mymodule::string_or_default('password', $parameters, $defaults)
$enable_password = mymodule::string_or_default('enable_password', $parameters, $defaults)
```